components/BottomNavigation/tests/unit/Theming/MDCBottomNavigationThemingTests.m

```diff
@@ -0,0 +1,180 @@
+  containerScheme.colorScheme = [self customColorScheme];
+
+  // When
+  [bottomNav applyPrimaryThemeWithScheme:containerScheme];
+
+  // Then
+  [self helperTextPrimaryThemeColorForColorScheme:containerScheme.colorScheme];
+  [self helperTestItemTitleFontEqualsFont:containerScheme.typographyScheme.caption];
+  [self helperTestNonsubsytemValuesForBottomNavigationBar:bottomNav];
+}
+
+- (void)helperTextPrimaryThemeColorForColorScheme:(id<MDCColorScheming>)colorScheme {
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  XCTAssertEqualObjects(bottomNav.barTintColor, colorScheme.primaryColor);
+  XCTAssertEqualObjects(bottomNav.selectedItemTintColor, colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects(bottomNav.selectedItemTitleColor, colorScheme.onPrimaryColor);
+  XCTAssertEqualObjects(
+      bottomNav.unselectedItemTintColor,
+      [colorScheme.onPrimaryColor colorWithAlphaComponent:kUnselectedPrimaryAlpha]);
+}
+
+- (void)testSurfaceThemeWithDefaultScheme {
+  // Given
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  MDCContainerScheme *containerScheme = [[MDCContainerScheme alloc] init];
+
+  // When
+  [bottomNav applySurfaceThemeWithScheme:containerScheme];
+
+  // Then
+  [self helperTextSecondaryThemeColorForColorScheme:containerScheme.colorScheme];
+  [self helperTestItemTitleFontEqualsFont:containerScheme.typographyScheme.caption];
+  [self helperTestNonsubsytemValuesForBottomNavigationBar:bottomNav];
+}
+
+- (void)testSurfaceThemeWithCustomContainerScheme {
+  // Given
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  MDCContainerScheme *containerScheme = [[MDCContainerScheme alloc] init];
+  containerScheme.typographyScheme = [self customTypographyScheme];
+  containerScheme.colorScheme = [self customColorScheme];
+
+  // When
+  [bottomNav applySurfaceThemeWithScheme:containerScheme];
+
+  // Then
+  [self helperTextSecondaryThemeColorForColorScheme:containerScheme.colorScheme];
+  [self helperTestItemTitleFontEqualsFont:containerScheme.typographyScheme.caption];
+  [self helperTestNonsubsytemValuesForBottomNavigationBar:bottomNav];
+}
+
+- (void)helperTextSecondaryThemeColorForColorScheme:(id<MDCColorScheming>)colorScheme {
+  MDCBottomNavigationBar *bottomNav = self.bottomNavigationBar;
+  XCTAssertEqualObjects(bottomNav.barTintColor, colorScheme.surfaceColor);
+  XCTAssertEqualObjects(bottomNav.selectedItemTintColor, colorScheme.primaryColor);
+  XCTAssertEqualObjects(bottomNav.selectedItemTitleColor, colorScheme.primaryColor);
+  XCTAssertEqualObjects(
+      bottomNav.unselectedItemTintColor,
+      [colorScheme.onSurfaceColor colorWithAlphaComponent:kUnselectedSurfaceAlpha]);
+}
+
+- (void)helperTestItemTitleFontEqualsFont:(UIFont *)font {
+  XCTAssertTrue([self.bottomNavigationBar.itemTitleFont mdc_isSimplyEqual:font],
+                @"%@ is not equal to %@", self.bottomNavigationBar.itemTitleFont, font);
```

Comment:
Tip: Nit: Please prefer using a format similar to the built-ins. `@"(%@) is not equal to (%@)"`. The parentheses are handy in identifying where the test/reference objects begin and end.

Reply:
Done